catalog/plugin/catalog-plugin-metacardbackup-filestorage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/filestorage/MetacardBackupFileStorage.java

```diff
@@ -0,0 +1,210 @@
+                DESCRIBABLE_PROPERTIES_FILE)) {
+            describableProperties.load(properties);
+        } catch (IOException e) {
+            LOGGER.info(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    @Override
+    public String getVersion() {
+        return (String) describableProperties.get(VERSION);
+    }
+
+    @Override
+    public String getTitle() {
+        return (String) describableProperties.get(TITLE);
+    }
+
+    @Override
+    public String getDescription() {
+        return (String) describableProperties.get(DESCRIPTION);
+    }
+
+    @Override
+    public String getOrganization() {
+        return (String) describableProperties.get(ORGANIZATION);
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public void deleteData(String id) throws IOException, MetacardBackupException {
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new MetacardBackupException(
+                    "Unable to delete stored data; no output directory specified.");
+        }
+
+        deleteBackupIfPresent(id);
+    }
+
+    public void storeData(String id, byte[] data) throws IOException, MetacardBackupException {
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new MetacardBackupException("Unable to store data; no output directory specified.");
+        }
+
+        if (data == null) {
+            throw new MetacardBackupException("No data to store");
+        }
+
+        Path metacardPath = getMetacardDirectory(id);
+        if (metacardPath == null) {
+            String message = String.format("Unable to create metacard path directory for %s", id);
+            LOGGER.debug(message);
+            throw new MetacardBackupException(message);
+        }
+
+        try {
+            Path parent = metacardPath.getParent();
+            if (parent != null) {
+                Files.createDirectories(parent);
+            }
+            Files.createFile(metacardPath);
+        } catch (IOException e) {
+            LOGGER.debug("Unable to create backup file {}.  File may already exist.",
+                    metacardPath,
+                    e);
+        }
+
+        try (OutputStream outputStream = new FileOutputStream(metacardPath.toFile())) {
+            IOUtils.write(data, outputStream);
+        }
+    }
+
+    public void setOutputDirectory(String outputDirectory) {
+        this.outputDirectory = outputDirectory;
+    }
+
+    public String getOutputDirectory() {
+        return outputDirectory;
+    }
+
+    public void refresh(Map<String, Object> properties) {
+        Object outputDirectory = properties.get(OUTPUT_DIRECTORY_PROPERTY);
+        if (outputDirectory instanceof String && StringUtils.isNotBlank((String) outputDirectory)) {
+            this.outputDirectory = (String) outputDirectory;
+            LOGGER.debug("Updating {} with {}", OUTPUT_DIRECTORY_PROPERTY, outputDirectory);
```

Comment:
Removed this as we don't need to log when configuration changes, as it's reflected in the AdminUI